executor/write.go

```diff
@@ -2090,3 +2091,17 @@ func (e *UpdateExec) Close() error {
 func (e *UpdateExec) Open(ctx context.Context) error {
 	return e.SelectExec.Open(ctx)
 }
+
+func batchDMLIfNeeded(ctx sessionctx.Context, count int) error {
+	if count%ctx.GetSessionVars().DMLBatchSize == 0 && count != 0 {
+		if err := ctx.StmtCommit(); err != nil {
+			return ErrBatchDMLFail.GenByArgs(err)
```

Comment:
`StmtRollback` here when error is not nil, otherwise the statement changes will be add to the next transaction. @jackysp

Reply:
I think there is a cleanup in `StmtCommit`
https://github.com/pingcap/tidb/blob/7aefd5067ecddefb61d9def7a112e90d590bde0f/session/txn.go#L339
Will it happen?